tfcompile_study/utils.py

```diff
@@ -0,0 +1,79 @@
+from tf2xla_pb2 import Config, TensorId
+import tensorflow as tf
+import subprocess
+
+def config_from_graph(input_tensors,
+                      output_tensors,
+                      graph_def,
+                      trainable_vars_names,
+                      is_training):
+    """
+    Generate config based on graphdef
+    Args:
+        input_tensors: inputs for the model function (eg. features, labels etc.)
+        output_tensors: outputs from the model function (eg. loss)
+        graph_def: GraphDef of the model.
+        trainable_vars_names: trainable variable names from according to tensorflow graph
+        is_training: If Batchnorm/Dropout training is False
+    Returns config proto for tfcompile
+    """
+    config = Config()
+    num_nodes = len(graph_def.node)
```

Comment:
```python
    input_tensor_names = [tensor.op.name for tensor in input_tensors]
    output_tensor_names = [tensor.op.name for tensor in output_tensors]

    def _shape_type(n):
        return {
            "shape": n.attr["shape"].shape,
            "type": n.attr["dtype"].type
        }

    for node in graph_def.node:
        if node.name in input_tensor_names:
            config.feed.add(id=TensorId(node_name=node.name),
                            **_shape_type(node))

        if node.name in output_tensor_names:
            config.fetch.add(id=TensorId(node_name=node.name))

        if node.op == "VarHandleOp":
            trainable = node.name in trainable_vars_names
            read_write = trainable or is_training

            var_args = {
                "node_name": node.name,
                "readonly": not read_write,
                **_shape_type(node),
            }
            config.variable.add(**var_args)

    return config
```

Reply:
replaced with the above version :). Thanks!!